File: app/src/main/java/dk/dtu/isaacirani/kirurgisksimulator/ScenarioRepository.java
package dk.dtu.isaacirani.kirurgisksimulator;
        import com.google.firebase.database.DatabaseReference;
        import com.google.firebase.database.FirebaseDatabase;
        import java.util.ArrayList;

        import dk.dtu.isaacirani.kirurgisksimulator.models.Scenario;

public class ScenarioRepository {
    DatabaseReference mDatabase = FirebaseDatabase.getInstance().getReference();
    ArrayList<Scenario> scenarioList = new ArrayList<>();


    public ArrayList<Scenario> getScenarioList() {

        return scenarioList;
    }


    public void createScenario(Scenario scenario) {
        String key = mDatabase.child("Scenarios").push().getKey();
        mDatabase.child("Scenarios").child(key).setValue(scenario);
    }

    public void updateScenario() {


    }

    public void deleteScenario() {

    }

}

File: app/src/main/java/dk/dtu/isaacirani/kirurgisksimulator/repositories/ScenarioRepository.kt
package dk.dtu.isaacirani.kirurgisksimulator.repositories

import android.util.Log
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.database.ValueEventListener
import dk.dtu.isaacirani.kirurgisksimulator.models.Scenario

public class ScenarioRepository {

    var mDatabase = FirebaseDatabase.getInstance().reference
    var scenarioRef = mDatabase.child("Scenarios")

    fun loadScenarios(callBack: (ArrayList<Scenario>) -> Unit) {
        scenarioRef.addValueEventListener(object : ValueEventListener {
            override fun onDataChange(dataSnapshot: DataSnapshot) {
                var scenarios: ArrayList<Scenario> = ArrayList();

                for(itemSnapShot: DataSnapshot in dataSnapshot.children) {
                    var entry: Scenario = itemSnapShot.getValue(Scenario::class.java)!!
                    scenarios.add(entry)
                }

                callBack(scenarios)
            }

            override fun onCancelled(error: DatabaseError) {
                // Failed to read value
                Log.w("ERROR", "Failed to read value.", error.toException())
            }
        })
    }

    fun createScenario(entry: Scenario) : String{
        val id = scenarioRef.push().key!!
        scenarioRef.child(id).setValue(entry)
        return id;
    }

}